### PR TITLE
Use Signer trait without accessing parties directly

### PR DESCRIPTION
**Pull request by xoloki**
In wtfrost v4, we now have a full Signer trait which can be used to do all operations without accessing the encapsulated parties; hence the parties are no longer pub fields.

This PR updates core-eng to use the Signer trait exclusively, which will allow us to easily move between vanilla FROST (v1) and WTF (v2).

### CODE REVIEW

File: frost-coordinator/src/coordinator.rs [hunk 1]
```diff
@@ -166,22 +166,27 @@ where
...
+                .sign(&self.network_private_key)
+                .expect("Failed to sign NonceRequest"),
             msg: MessageTypes::NonceRequest(nonce_request),
         };
 
-        debug!("dkg_id #{}. NonceRequest sent.", self.current_dkg_id);
+        info!(
+            "dkg_id #{} sign_id #{} sign_nonce_id #{}. NonceRequest sent",
+            self.current_dkg_id, self.current_sign_id, self.current_sign_nonce_id
+        );
```
Comment (netrome) [inline frost-coordinator/src/coordinator.rs:178]: Let's remember to change this back to `debug` once we have proper env logging
```suggestion
        // TODO(#286): Change back to debug level
        info!(
            "dkg_id #{} sign_id #{} sign_nonce_id #{}. NonceRequest sent",
            self.current_dkg_id, self.current_sign_id, self.current_sign_nonce_id
        );
```

File: frost-coordinator/src/coordinator.rs [hunk 2]
```diff
@@ -203,57 +208,74 @@ where
     fn compute_aggregate_nonce(&mut self, msg: &[u8]) -> Result<Point, Error> {
         info!("Computing aggregate nonce...");
         self.collect_nonces()?;
-        let ids: Vec<usize> = self.public_nonces.keys().map(|i| *i as usize).collect();
-        let nonces: Vec<PublicNonce> = self
+        //let ids: Vec<usize> = self.public_nonces.keys().map(|i| *i as usize).collect();
+        // XXX this needs to be key_ids for v1 and signer_ids for v2
```
Comment (netrome) [inline frost-coordinator/src/coordinator.rs:212]: Nit: Clean up

File: frost-coordinator/src/coordinator.rs [hunk 3]
```diff
@@ -284,7 +306,7 @@ where
         }
 
         // make an array of dkg public share polys for SignatureAggregator
-        debug!(
+        info!(
```
Comment (netrome) [inline frost-coordinator/src/coordinator.rs:309]:
```suggestion
        // TODO(#286): Change back to debug
        info!(
```

File: frost-coordinator/src/coordinator.rs [hunk 4]
```diff
@@ -295,7 +317,7 @@ where
             .map(|ps| ps.public_share.clone())
             .collect();
 
-        debug!(
+        info!(
```
Comment (netrome) [inline frost-coordinator/src/coordinator.rs:320]:
```suggestion
        // TODO(#286): Change back to debug
        info!(
```

**Review by netrome — APPROVED** [review]
lgtm. Just some nits about logging.

File: frost-coordinator/src/coordinator.rs (hunk 1 above)
Comment (xoloki) [reply frost-coordinator/src/coordinator.rs:178]: Done.

File: frost-coordinator/src/coordinator.rs (hunk 2 above)
Comment (xoloki) [reply frost-coordinator/src/coordinator.rs:212]: Done.

File: frost-coordinator/src/coordinator.rs (hunk 3 above)
Comment (xoloki) [reply frost-coordinator/src/coordinator.rs:309]: Done.

**Comment by xoloki** [comment]
> lgtm. Just some nits about logging.

All fixed.

**Comment by netrome** [comment]
> > lgtm. Just some nits about logging.
> 
> All fixed.

That was quick, thanks! :rocket:

**Comment by xoloki** [comment]
> > > lgtm. Just some nits about logging.
> > 
> > 
> > All fixed.
> 
> That was quick, thanks! 🚀

It wasn't a big ask lol

**Review by stjepangolemac — APPROVED** [review]

**Pull request merged**